webapi/src/main/scala/org/knora/webapi/responders/Responder.scala

```diff
@@ -174,14 +173,23 @@ abstract class Responder(responderData: ResponderData) extends LazyLogging {
     * @return IRI of the entity.
     */
   protected def checkOrCreateEntityIri(entityIri: Option[SmartIri], iriFormatter: => IRI): Future[IRI] = {
+
     entityIri match {
-      case Some(customResourceIri) =>
+      case Some(customEntityIri: SmartIri) =>
+        val entityIriAsString = customEntityIri.toString
         for {
-          result <- stringFormatter.checkIriExists(customResourceIri.toString, storeManager)
+
+          result <- stringFormatter.checkIriExists(entityIriAsString, storeManager)
           _ = if (result) {
-            throw DuplicateValueException(s"IRI: '${customResourceIri.toString}' already exists, try another one.")
+            throw DuplicateValueException(s"IRI: '$entityIriAsString' already exists, try another one.")
           }
-        } yield customResourceIri.toString
+          // Check that given entityIRI ends with a UUID
```

Comment:
Shouldn't this be already checked by StringFormater when the SmartIri is created? StringFormater shouldn't allow the creation of invalid IRIs.

Reply:
Currently, we are only dealing with custom IRIs and only want to check that custom IRIs end with UUID. Changing the logic of `toSmartIri` requires all the IRIs (in test data, etc.)  to end with UUID. We are not there yet, it can be added to the PR about changing entity IRs.

Reply:
great thanks